core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushTopNIntoTableScan.java

```diff
@@ -43,10 +44,10 @@
 {
     private static final Capture<TableScanNode> TABLE_SCAN = newCapture();
 
-    // Currently the rule is applied at the optimization phase where PARTIAL and FINAL TopNNode do not exist.
-    // The rule can be further made to work with PARTIAL and FINAL if needed.
+    // Rule is executed in two planning phases. Initially we try to pushdown SINGLE TopN into
+    // table scan. If that fails, we repeat the exercise for PARTIAL TopN nodes after SINGLE -> PARTIAL/FINAL split.
     private static final Pattern<TopNNode> PATTERN = topN()
-            .matching(node -> node.getStep().equals(Step.SINGLE))
+            .matching(node -> EnumSet.of(Step.SINGLE, Step.PARTIAL).contains(node.getStep()))
```

Comment:
Add a comment at https://github.com/trinodb/trino/blob/b2f1f698ee8978aa4d6c7b54c147ad7fe35aa0e4/core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushTopNIntoTableScan.java#L83-L85 capturing the some reasoning (or a hint that there _was_ a reasoning) why removing PARTIAL TopN, while leaving FINAL TopN in the plan, is an OK thing to do

Reply:
Added sth